internal/factsengine/gatherers/sapservices.go

```diff
@@ -0,0 +1,187 @@
+package gatherers
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+type SapServicesStartupKind string
+
+const (
+	SapServicesSystemdStartup  SapServicesStartupKind = "systemctl"
+	SapServicesSapstartStartup SapServicesStartupKind = "sapstartsrv"
+	sapServicesDefaultPath                            = "/usr/sap/sapservices"
+	SapServicesGathererName                           = "sapservices"
+)
+
+// nolint:gochecknoglobals
+var (
+	SapServicesParsingError = entities.FactGatheringError{
+		Type:    "sap-services-parsing-error",
+		Message: "error parsing the sap services file",
+	}
+	SapServicesFileError = entities.FactGatheringError{
+		Type:    "sap-services-parsing-error",
+		Message: "error reading the sap services file",
+	}
+	SapstartSIDExtractionPattern = regexp.MustCompile(`(?s)pf=([^[:space:]]+)/(.*?)_.*_.*`)
```

Comment:
Do you group the `space` thing for any reason? I don't think it is used
By the way, wouldn't be the usage of `.+` more appropriate?
I mean `.*` would match even with empty string

Reply:
I made the regexp on regex101 as far as I understood it's not a capture but just a more explicit way to express the things, I don't find also the group in the golang extraction

Reply:
Well, in `extractSIDFromSapstartService` you look for a 3 elements size match (which includes, the whole text, 2 groups)